community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/CypherCompiler.scala

```diff
@@ -58,7 +58,7 @@ case class PreParsedQuery(statement: String, rawStatement: String, version: Cyph
       case _ => s"strategy=${updateStrategy.name}"
     }
 
-    s"CYPHER ${version.name} $plannerInfo $runtimeInfo $updateStrategyInfo $statement".replaceAll("\\s+", " ")
+    s"CYPHER ${version.name} $plannerInfo $runtimeInfo $updateStrategyInfo $statement"
```

Comment:
I guess this change fixes the problem with string literals, but it will have the drawback the two identical queries which only differ in a white-space will be planned separately and take 2 query cache slots. 

It's of course better to return the correct results, that to use less planning time, but if we were to solve both we probably would need to parse the query and rebuild from AST.